xbout/plotting/animate.py

```diff
@@ -45,12 +47,27 @@ def animate_imshow(data, animate_over='t', x='x', y='y', animate=True,
     variable = data.name
 
     # Check plot is the right orientation
-    t_read, y_read, x_read = data.dims
-    if (x_read is x) & (y_read is y):
-        pass
-    elif (x_read is y) & (y_read is x):
+    spatial_dims = list(data.dims)
+
+    try:
+        spatial_dims.remove(animate_over)
+    except ValueError:
+        raise ValueError("Dimension animate_over={} is not present in the data"
+                         .format(animate_over))
+
+    try:
```

Comment:
Maybe instead of this `try...except` around everything, it would be better to check the that `len(spatial_dims) == 3`, then `try...except` only around the transpose?

Reply:
The `try...except` is also to catch if `spatial_dims.remove(y)` or `spatial_dims.remove(x)` throw because wrong dimensions were passed. We should probably check that `len(data.dims) == 3` somewhere, or `transpose` will throw `ValueError` leading to a message that isn't helpful if, e.g. you forgot an `isel(z=0)`.

Reply:
> The try...except is also to catch if spatial_dims.remove(y) or spatial_dims.remove(x) throw because wrong dimensions were passed.

Yes, but they should really have their own specific error messages, so their own try except statements?